### PR TITLE
Committing batch and schedule class Assignment

### PR DESCRIPTION
**Pull request by Vardhman-Fexle-Trainee**

### CODE REVIEW

File: classes/LeadProcessorTest.cls [hunk 1]
```diff
@@ -0,0 +1,32 @@
+@IsTest
+public with sharing class LeadProcessorTest {
```
Comment (shubhamAgarwal242) [inline classes/LeadProcessorTest.cls]: @Vardhman-Fexle-Trainee, why you have added with sharing here?

File: classes/LeadProcessor.cls [hunk 2]
```diff
@@ -0,0 +1,36 @@
+global class LeadProcessor implements Database.Batchable<SObject>, Schedulable {
+    
+    global  Integer recordProcesses = 0; 
+    
+    global Database.QueryLocator start(Database.BatchableContext bc){
+        
+        return Database.getQueryLocator('SELECT Id, LeadSource FROM Lead' );
```
Comment (shubhamAgarwal242) [inline classes/LeadProcessor.cls]: @Vardhman-Fexle-Trainee, we should add the check for LeadSource != 'Dreamforce' in this query after that we can remove the check from execute method

File: classes/LeadProcessorTest.cls [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        Test.startTest();
+        
+        //insertLeads();
+        LeadProcessor leadpro = new LeadProcessor();
+        Id batchId = Database.executeBatch(leadpro);
+        Test.stopTest();
+        
+        List<Lead> leadList = [SELECT LeadSource FROM Lead WHERE LeadSource = 'Dreamforce'];
+        
+        System.debug(leadList.size());
```
Comment (shubhamAgarwal242) [inline classes/LeadProcessorTest.cls]: @Vardhman-Fexle-Trainee, Remove the debugs, Any debug should not deploy to production

File: classes/LeadProcessorTest.cls [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        System.assertEquals(200, leadList.size(), 'List is not returning 200 records') ;
```
Comment (shubhamAgarwal242) [inline classes/LeadProcessorTest.cls]: @Vardhman-Fexle-Trainee, Please user Asser class intead of system.assertEquals

**Review by shubhamAgarwal242 — CHANGES_REQUESTED** [review]

File: classes/LeadProcessorTest.cls (hunk 1 above)
Comment (Vardhman-Fexle-Trainee) [reply classes/LeadProcessorTest.cls]: As it is added by Vs code by default and I forgot to change it.
Right now I am removing it.

File: classes/LeadProcessorTest.cls (hunk 4 above)
Comment (Vardhman-Fexle-Trainee) [reply classes/LeadProcessorTest.cls]: Sure, I will add assert class method here.

File: classes/LeadProcessorTest.cls (hunk 3 above)
Comment (Vardhman-Fexle-Trainee) [reply classes/LeadProcessorTest.cls]: Okay, I will remove it.

File: classes/LeadProcessor.cls (hunk 2 above)
Comment (Vardhman-Fexle-Trainee) [reply classes/LeadProcessor.cls]: Okay, so I have to use where clause.

**Review by shubhamAgarwal242 — APPROVED** [review]

**Pull request merged**